### DISCUSSION TITLE
Feature Request: Add a hash function for strs similar to hashFiles

### DISCUSSION THREAD

**Discussion by leighmcculloch** [community/community#16615] [Actions]
It would be really useful if there was a `hash` function for SHA256 hashing strings, similar to how there is a `hashFiles` function that hashes files.

It would primarily be useful for hashing strings for cache keys and other inputs that need to be short but uniquely defined based off some larger non-file input.

**Comment by jsoref** [comment]
I was about to file this.

I'd like to take what's essentially a hash of this input:
https://github.com/check-spelling/check-spelling/blob/d7cd2973c513e84354f9d6cf50a6417a628a78ce/action.yml#L72-L141

I really don't want to create a key which is the `toJSON` of that input (it's going to be really long and very ugly), but at present, my choices are:

1. use `toJSON` on those inputs (technically there are a couple of inputs I need...)
2. create an extra step that takes that inputs and creates my own hash spitting it out as an output, and then using that output as the value for where I'd want to use `hash(inputs.check_extra_dictionaries...)`

**Reply by jsoref** [reply]
For kicks, I just tried, and here's what I got:
```
Error: Key Validation Error: check-spelling-dictionaries-{"cspell": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20220816/dictionaries/"}-"cspell:software-terms/src/software-terms.txt cspell:php/php.txt"-"cspell:ada/ada.txt cspell:aws/aws.txt cspell:bash/bash-words.txt cspell:clojure/src/clojure.txt cspell:companies/src/companies.txt cspell:cpp/src/compiler-clang-attributes.txt cspell:cpp/src/compiler-gcc.txt cspell:cpp/src/compiler-msvc.txt cspell:cpp/src/cpp.txt cspell:cpp/src/ecosystem.txt cspell:cpp/src/lang-jargon.txt cspell:cpp/src/lang-keywords.txt cspell:cpp/src/people.txt cspell:cpp/src/stdlib-c.txt cspell:cpp/src/stdlib-cerrno.txt cspell:cpp/src/stdlib-cmath.txt cspell:cpp/src/stdlib-cpp.txt cspell:cpp/src/template-strings.txt cspell:cryptocurrencies/cryptocurrencies.txt cspell:csharp/csharp.txt cspell:css/css.txt cspell:dart/src/dart.txt cspell:django/django.txt cspell:django/requirements.txt cspell:docker/src/docker-words.txt cspell:dotnet/dotnet.txt cspell:elixir/elixir.txt cspell:filetypes/filetypes.txt cspell:fonts/fonts.txt cspell:fullstack/fullstack.txt cspell:golang/go.txt cspell:haskell/haskell.txt cspell:html-symbol-entities/entities.txt cspell:html/html.txt cspell:java/java.txt cspell:latex/latex.txt cspell:latex/samples/sample-words.txt cspell:lorem-ipsum/dictionary.txt cspell:lua/lua.txt cspell:mnemonics/src/mnemonics.txt cspell:monkeyc/src/monkeyc_keywords.txt cspell:node/additional_words.txt cspell:node/node.txt cspell:node/src/node-old.txt cspell:npm/npm.txt cspell:php/php.txt cspell:powershell/powershell.txt cspell:public-licenses/src/additional-licenses.txt cspell:public-licenses/src/generated/public-licenses.txt cspell:python/src/additional_words.txt cspell:python/src/common/extra.txt cspell:python/src/python/python-lib.txt cspell:python/src/python/python.txt cspell:r/src/r.txt cspell:ruby/ruby.txt cspell:rust/rust.txt cspell:scala/scala.txt cspell:software-terms/src/network-os.txt cspell:software-terms/src/network-protocols.txt cspell:software-terms/src/software-terms.txt cspell:software-terms/src/software-tools.txt cspell:sql/src/sql.txt cspell:sql/src/tsql.txt cspell:swift/src/swift.txt cspell:typescript/typescript.txt cspell:win32/src/win32.txt" cannot be larger than 512 characters.
```

So, yes, it'd be really appreciate if such a function exists since there's clearly a rule that prevents users from not doing this.

**Comment by iamsuiux** [comment]
There are many realizations of hash functions written in JS. For example:
SHA-1: http://www.webtoolkit.info/javascript-sha1.html
SHA-256: http://www.webtoolkit.info/javascript-sha256.html
MD5: http://www.webtoolkit.info/javascript-md5.html

**Reply by jsoref** [reply]
And we shouldn't have to copy one (or miscopy one) just in order to perform a simple task, e.g. providing input to actions/cache.

The github runner expressions area is not designed to let you add your own functions, it's designed to evaluate simple expressions including calling whichever special functions the runner exposes, and this is not one of the functions it exposes.
